burn-tensor/src/tensor/api/base.rs

```diff
@@ -257,6 +257,40 @@ where
         self.reshape(shape)
     }
 
+    /// Creates a new tensor with a dimension of size one inserted at the specified position.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Tensor, Shape};
+    ///
+    /// fn example<B: Backend>() {
+    ///     let tensor = Tensor::<B, 2>::ones(Shape::new([3, 3]));
+    ///     let tensor = tensor.unsqueeze_dim::<3>(1);
```

Comment:
Could we use this style instead?

```rust
let tensor:Tensor<B, 3> = tensor.unsqueeze_dim(1);
```

`<3>` stands out compared to other Burn examples.